sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -198,8 +225,16 @@ static void _process_coap_pdu(sock_udp_t *sock, sock_udp_ep_t *remote,
         /* normal request */
         else if (coap_get_type(&pdu) == COAP_TYPE_NON
                 || coap_get_type(&pdu) == COAP_TYPE_CON) {
-            size_t pdu_len = _handle_req(&pdu, _listen_buf, sizeof(_listen_buf),
+            size_t pdu_len;
+
+            if (truncated) {
+                /* TBD: Set a Size1 */
+                pdu_len = gcoap_response(&pdu, _listen_buf, sizeof(_listen_buf), COAP_CODE_REQUEST_ENTITY_TOO_LARGE);
+            } else {
+                pdu_len = _handle_req(&pdu, _listen_buf, sizeof(_listen_buf),
                                             remote);
```

Comment:
```suggestion
                pdu_len = _handle_req(&pdu, _listen_buf, sizeof(_listen_buf),
                                      remote);
```

I cannot easily see in the web interface whether this would fit 100 chars. If it does, I would encourage to not break the line.

Reply:
It did fit in the 100-char limit; I took the liberty to squash this all right away.